addons/binding/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/handler/PLCCommonHandler.java

```diff
@@ -0,0 +1,279 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.plclogo.handler;
+
+import static org.openhab.binding.plclogo.PLCLogoBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.plclogo.PLCLogoBindingConstants;
+import org.openhab.binding.plclogo.PLCLogoBindingConstants.Layout;
+import org.openhab.binding.plclogo.internal.PLCLogoClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PLCCommonHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Alexander Falkenstern - Initial contribution
+ */
+@NonNullByDefault
+public abstract class PLCCommonHandler extends BaseThingHandler {
+
+    public static final int INVALID = Integer.MAX_VALUE;
+
+    private final Logger logger = LoggerFactory.getLogger(PLCCommonHandler.class);
+
+    private Map<String, @Nullable State> oldValues = new HashMap<>();
+    private @Nullable PLCLogoClient client;
+    private @Nullable String family;
+
+    /**
+     * Constructor.
+     */
+    public PLCCommonHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        scheduler.execute(new Runnable() {
+            @Override
+            public void run() {
+                doInitialization();
+            }
+        });
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Dispose LOGO! common block handler.");
+        super.dispose();
+
+        final ThingBuilder tBuilder = editThing();
+        for (final Channel channel : getThing().getChannels()) {
+            tBuilder.withoutChannel(channel.getUID());
+        }
+        updateThing(tBuilder.build());
+
+        synchronized (oldValues) {
+            oldValues.clear();
+        }
+
+        client = null;
+        family = null;
+    }
+
+    @Override
+    protected @Nullable Bridge getBridge() {
+        final Bridge bridge = super.getBridge();
+        Objects.requireNonNull(bridge, "PLCCommonHandler: Bridge may not be null.");
```

Comment:
Normally you would use this to validate input, now this will give a pretty obscure result, its annotated `@Nullable` but it will either return a value or a NPE (yes, that is the body of `requireNonNull`)

What are you aiming for.